service-registry/src/test/java/io/servicecomb/serviceregistry/TestRegistry.java

```diff
@@ -396,27 +402,128 @@ public void onMicroserviceInstanceChanged(MicroserviceInstanceChangedEvent chang
         Assert.assertTrue(status);
     }
 
-    @Test
-    public void testRegistryUtilGetPublishAddress() {
+    public void testRegistryUtilGetPublishAddress(@Mocked InetAddress ethAddress) {
+        new Expectations(NetUtils.class) {
+            {
+                NetUtils.getHostAddress();
+                result = "1.1.1.1";
+            }
+        };
         String address = RegistryUtils.getPublishAddress();
-        Assert.assertNotNull(address);
-        System.setProperty("cse.service.publishAddress", "{eth0}");
-        address = RegistryUtils.getPublishAddress();
-        Assert.assertNotNull(address);
+        Assert.assertEquals("1.1.1.1", address);
+
+        new Expectations(DynamicPropertyFactory.getInstance()) {
+            {
+                DynamicPropertyFactory.getInstance().getStringProperty("cse.service.publishAddress", "");
```

Comment:
It could be better if we use constant value for the "cse.service.publishAddress".